File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/orders/repository/OrderRepository.java
package com.portfolio.auctionmarket.domain.orders.repository;

import com.portfolio.auctionmarket.domain.orders.entity.Order;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface OrderRepository extends JpaRepository<Order, Long> {

    @EntityGraph(attributePaths = {"auction", "auction.product", "auction.product.image"})
    Page<Order> findAllByBuyer_UserId(Long userId, Pageable pageable);

    Optional<Order> findByAuction_AuctionId(Long auctionAuctionId);
}

File: AuctionMarket/auction-domain/src/main/kotlin/org/example/auctioncommon/domain/order/repository/OrderRepository.kt
package org.example.auctioncommon.domain.order.repository

import org.example.auctioncommon.domain.order.entity.Order
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface OrderRepository : JpaRepository<Order, Long> {
    @EntityGraph(attributePaths = ["auction", "auction.product", "auction.product.image"])
    fun findAllByBuyer_UserId(userId: Long, pageable: Pageable): Page<Order>

    fun findByAuction_AuctionId(auctionAuctionId: Long): Order?
}
